python/lsst/ts/salobj/domain.py

```diff
@@ -31,16 +31,65 @@
 from lsst.ts import idl
 from . import base
 
-DDS_READ_QUEUE_LEN = 100
-"""Length of DDS read queue
+MAX_RANDOM_HOST = (1 << 31) - 1
 
-Warning: this must be equal to or greater than the queue length in the
-OpenSplice configuration file (pointed to by $OSPL_URI).
-This information is not available from ``dds`` objects, so I set queue depth
-instead of using the value specified in the QoS XML file.
-"""
 
-MAX_RANDOM_HOST = (1 << 31) - 1
+class QosSet:
+    """QoS profiles for topic, reader and writer.
+
+    Parameters
+    ----------
+    qos_path : `str` or `pathlib.Path`
+        Path to QoS XML file.
+    profile_name : `str`
+        Profile name; one of "Command", "Event", or "Telemetry".
+
+    Attributes
+    ----------
+    profile_name : `str`
+        Profile name; one of "Command", "Event", or "Telemetry".
+    volatile : `bool`
+        Is this category of topics volatile?
+        Warning: this is determined from ``profile_name``
+        (because there is no way to read it from the QoS)
+        and that logic needs to be updated if the QoS file changes.
+        # Length of DDS read queues.
+        # This is primarily used to warn of the queues filling up.
+        # Those warnings will be more accurate if this matches the value in the
+        # Quality of Service configuration file pointed to by $LSST_DDS_QOS.
+        # (Note: this information is not available from ``dds`` objects.)
+    dds_read_queue_len : `int`
+        Length of DDS read queues.
+        This is primarily used to warn of the queues filling up.
+        Those warnings will be more accurate if this matches the value in the
+        Quality of Service configuration file pointed to by $LSST_DDS_QOS.
+        (Unfortunately this information is not available from ``dds`` objects.)
+
+    Notes
+    -----
+    The following QoS should be created elsewhere:
+    * publisher and subscriber: these depend on the DDS partition name,
+      which is only read when creating the SalInfo object.
+       QoS does not depend on the topic type,
+    * participant: this should not depend on the profile name,
+      and there's no point making 3 of them.
+    """
+
+    def __init__(self, qos_path, profile_name):
+        self.profile_name = profile_name
+        self.volatile = dict(
+            AckcmdProfile=True,
+            CommandProfile=True,
+            EventProfile=False,
+            TelemetryProfile=True,
+        )[profile_name]
+
+        self.dds_read_queue_len = 100
```

Comment:
So, this may be a picky comment but can you consider changing the name of the attribute to something that make it clear it is not the actual length of the queue? maybe

```
self.dds_read_queue_warning_level
```

or something on those lines?

Also, I wonder if this should be an editable attribute. Maybe make this a `@property`, which would allow you to document it even better? 

```
@property
def dds_read_queue_warning_level(self):
    """ ....
    """
    return 100
```

Reply:
So, as I was saying in private, I think this information is actually accessible through python. 

If you do;

```
self.reader_qos.history.depth
```

you will get the size of the queue.

Reply:
Great suggestion. I have updated the code accordingly. Note that reader_qos.durability_service.history_depth is always reported as 0 rather than the correct value so, much as I'd like to use that value to warn for excessively large values of max_history, I can't. I filed a technical debt ticket for that.